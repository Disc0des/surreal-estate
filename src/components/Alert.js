import React from "react";
import "../styles/Alert.css";
import PropTypes from "prop-types";

function Alert({ message, success }) {
  return (
    <div className={`Alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
}

export default Alert;

Alert.defaultProps = {
  success: false,
};
Alert.propTypes = {
  message: PropTypes.string.isRequired,
  success: PropTypes.bool,
};
