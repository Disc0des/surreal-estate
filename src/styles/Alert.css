.Alert {
  width: 400px;
  height: 25px;
  align-self: center;
  font-family: 'Poppins', sans-serif;
  font-weight: 600;

}

.Alert, .alert-success {
  background-color: #3fd351;
  text-align: center;
  border-color: #217a37;
  border-style: solid;
  border-width: 2px;
}

.Alert .alert-error {
  background-color: #c73939;
}