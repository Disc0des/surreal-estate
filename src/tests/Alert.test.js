import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("alert bar", () => {
  it("displays an error message", () => {
    const { getByText } = render(<Alert message="Error!" />);
    expect(getByText(/Error/).textContent).toBe("Error!");
  });
  it("displays a success message", () => {
    const { getByText } = render(<Alert message="Success!" success />);
    expect(getByText(/Success/).textContent).toBe("Success!");
  });
  it("does not render if message prop is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
  it("renders an error when value of message is error", () => {
    const { asFragment } = render(<Alert message="Error!" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
  it("renders success when value of message is success", () => {
    const { asFragment } = render(<Alert message="Success!" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});
