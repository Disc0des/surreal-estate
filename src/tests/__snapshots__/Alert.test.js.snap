// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alert bar does not render if message prop is empty 1`] = `<DocumentFragment />`;

exports[`alert bar renders an error when value of message is error 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`alert bar renders success when value of message is success 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Success!
  </div>
</DocumentFragment>
`;
